File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        int l1=0,l2=0;
        ListNode *t1=headA,*t2=headB;
        while(t1){
            l1++;
            t1=t1->next;
        }

        while(t2){
            l2++;
            t2=t2->next;
        }
        t1=headA;t2=headB;
        if(l1>l2){
            while(t1 && l1!=l2){
                l1--;
                t1=t1->next;
            }
        }
        else if(l2>l1){
            while(t2 && l1!=l2){
                l2--;
                t2=t2->next;
            }
        }

        while(t1 && t2 && t1!=t2){
            t1=t1->next;
            t2=t2->next;
        }

        return t1==t2?t1:NULL;
    }
};

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        l1,l2 = headA,headB
        hAcpy,hBcpy = headA,headB 
        while l1 and l2 and l1!=l2: 
            l1=l1.next 
            l2=l2.next 
            if l1==None: 
                l1=hBcpy 
                hBcpy = None 
            if l2==None: 
                l2=hAcpy 
                hAcpy = None 
        
        return l1 